tensorflow_probability/python/experimental/stats/sample_stats.py

```diff
@@ -269,3 +273,169 @@ def __init__(self, shape=(), dtype=tf.float32):
         By default, the dtype is `tf.float32`.
     """
     super(RunningVariance, self).__init__(shape, event_ndims=0, dtype=dtype)
+
+
+RunningExpectationsState = collections.namedtuple(
+    'RunningExpectationsState', 'num_samples, expectation')
+
+
+class RunningExpectations(object):
+  """Computes expectations over arbitrary functions evaluated at samples.
+
+  To elaborate, any function that outputs a singular `Tensor` is accepted. If
+  one wishes to compute multiple expectations over the same samples, a
+  (possibly nested) collection of callables can be provided upon instantiation
+  (instead of only one). The resulting expectation calculations will mimic the
+  exact structure of the given callabes.
+
+  In computation, samples can be provided individually or in chunks. A
+  "chunk" of size M implies incorporating M samples into a single expectation
+  computation at once, which is more efficient than one by one. If more than one
+  callable is accepted and chunking is enabled, the chunked `axis` will define
+  chunking semantics for all callables.
+
+  `RunningExpectations` objects do not hold state information. That information,
+  which includes intermediate calculations, are held in a
+  `RunningExpectationsState` as returned via `initialize` and `update` method
+  calls.
+
+  `RunningExpectations` is meant to serve general streaming expectations.
+  For a specialized version that fits streaming over MCMC samples, see
+  `ExpectationsReducer` in `tfp.experimental.mcmc`.
+  """
+
+  def __init__(self, shape, callables, dtype=tf.float32):
```

Comment:
Why does this API need callables? I think the value transformation should be responsibility of the user code (just like it is with RunningCovariance).

Reply:
Yeah, that makes sense.